### DISCUSSION TITLE
report generation get_diffreport

### DISCUSSION THREAD

**Discussion by kottatip** [zubkov-andrei/pg_profile#36] [Q&A] [answered]
report generation, what am I doing wrong, tell me please....

select get_diffreport(tstzrange(now() - interval '4 hours',now() - interval '2 hours', now() - interval '1 hours', now()));
ERROR:  function tstzrange(timestamp with time zone, timestamp with time zone, timestamp with time zone, timestamp with time zone) does not exist
LINE 1: select get_diffreport(tstzrange(now() - interval '4 hours',n...
                              ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.

**Comment by zubkov-andrei** [comment, marked as answer]
Hello!

Building a differential report requires two time intervals. Function _tstzrange()_ gets two timestamps as an interval bounds, it can't be called with four arguments. You could use something like this:
```
select get_diffreport(
    tstzrange(now() - interval '4 hours',now() - interval '2 hours'),
    tstzrange(now() - interval '1 hours', now())
);
```

**Comment by kottatip** [comment]
and so it does not work, maybe this is due to the version (3.4)?
postgres=# select get_diffreport(
postgres(#     tstzrange(now() - interval '4 hours',now() - interval '2 hours'),
postgres(#     tstzrange(now() - interval '1 hours', now())
postgres(# );
ERROR:  function get_diffreport(tstzrange, tstzrange) does not exist
LINE 1: select get_diffreport(
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.

**Comment by zubkov-andrei** [comment]
Unfortunately _get_diffreport()_ function in case of time range parameters have no default for _server name_. You have to specify a server name, which is 'local' if you did not change it. Try this:
```
select get_diffreport(
    'local',
    tstzrange(now() - interval '4 hours',now() - interval '2 hours'),
    tstzrange(now() - interval '1 hours', now())
);
```

**Comment by kottatip** [comment]
yes it works, thank you very much ;-)
